### PR TITLE
[NCL-4795] Implement Brew push endpoints

### PR DESCRIPTION
**Pull request by thescouser89**
This PR replaces: #2457

### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java [hunk 1]
```diff
@@ -164,22 +168,32 @@ public void removeAttribute(int id, String key) {
...
     @Override
     public BuildPushResult getPushResult(int id) {
-        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+        return brewPusher.getBrewPushResult(id);
     }
 
     @Override
-    public Page<BuildPushResult> push(BuildPushRequest buildRecordPushRequest) {
-        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+    public Page<BuildPushResult> push(BuildPushRequest buildPushRequest) {
```
Comment (janinko) [inline rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java]: I realized this should probably return only `BuildPushResult`, no? There are not multiple results when pushing a single build.

File: facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java [hunk 2]
```diff
@@ -83,6 +98,65 @@ public void pushGroup(int id, String tagPrefix) {
...
+    }
+
+    public BuildPushResult getBrewPushResult(int buildId) {
+
+        BuildRecordPushResult latestForBuildRecord = buildRecordPushResultRepository.getLatestForBuildRecord(buildId);
+
+        if (latestForBuildRecord != null) {
+            return buildPushResultMapper.toDTO(latestForBuildRecord);
+        } else {
+            return null;
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java]: The mappers should be null safe

Comment (thescouser89) [reply facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java]: Noted!

File: rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java (hunk 1 above)
Comment (thescouser89) [reply rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java]: Right! I'll change the API to return only one result.

**Comment by thescouser89** [comment]
Repushed with changes @janinko ! :)

File: facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java [hunk 3]
```diff
@@ -83,6 +98,60 @@ public void pushGroup(int id, String tagPrefix) {
...
+        log.info("Push Results {}.", pushed.stream().map(r -> r.getId()).collect(Collectors.joining(",")));
+
+        Set<BuildPushResult> pushedResponse = pushed.stream()
+                .map(r -> BuildPushResult.builder()
+                        .id(Integer.parseInt(r.getId()))
+                        .buildId(buildId)
+                        .status(r.getStatus())
+                        .log(r.getMessage())
+                        .build())
+                .collect(Collectors.toSet());
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java]:
```suggestion
                .collect(Collectors.toList());
```

File: facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java [hunk 4]
```diff
@@ -83,6 +98,60 @@ public void pushGroup(int id, String tagPrefix) {
...
+        Set<Result> pushed = buildResultPushManager.push(
+                toPush,
+                currentUserToken(),
+                getCompleteCallbackUrl(),
+                buildPushRequest.getTagPrefix(),
+                buildPushRequest.isReimport());
+
+        log.info("Push Results {}.", pushed.stream().map(r -> r.getId()).collect(Collectors.joining(",")));
+
+        Set<BuildPushResult> pushedResponse = pushed.stream()
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java]:
```suggestion
        List<BuildPushResult> pushedResponse = pushed.stream()
```

File: facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java [hunk 5]
```diff
@@ -83,6 +98,60 @@ public void pushGroup(int id, String tagPrefix) {
...
+                .map(r -> BuildPushResult.builder()
+                        .id(Integer.parseInt(r.getId()))
+                        .buildId(buildId)
+                        .status(r.getStatus())
+                        .log(r.getMessage())
+                        .build())
+                .collect(Collectors.toSet());
+
+        return pushedResponse.toArray(new BuildPushResult[pushedResponse.size()])[0];
```
Comment (janinko) [inline facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java]:
```suggestion
        return pushedResponse.get(0);
```

**Comment by thescouser89** [comment]
Merged suggestions I think?

**Pull request merged**